x/ecocredit/client/tx.go

```diff
@@ -79,40 +79,85 @@ Parameters:
 	}
 }
 
+const (
+	FlagIssuer          string = "issuer"
+	FlagClassId         string = "class-id"
+	FlagStartDate       string = "start-date"
+	FlagEndDate         string = "end-date"
+	FlagProjectLocation string = "project-location"
+	FlagMetadata        string = "metadata"
+	FlagIssuance        string = "issuance"
+)
+
 func txCreateBatch() *cobra.Command {
-	return &cobra.Command{
-		Use:   "create-batch [issuer] [class_id] [start_date] [end_date] [metadata] [issuance]",
+	cmd := &cobra.Command{
+		Use:   "create-batch [--issuer issuer] [--class-id class_id] [--start-date start_date] [--end-date end_date] [--project-location project_location] [--metadata metadata] [--issuance issuance]",
 		Short: "Issues a new credit batch",
 		Long: `Issues a new credit batch.
 
 Parameters:
-  issuer:     issuer address
-  class_id:   credit class
-  start_date: The beginning of the period during which this credit batch was
-              quantified and verified. Format: yyyy-mm-dd.
-  end_date:   The end of the period during which this credit batch was
-              quantified and verified. Format: yyyy-mm-dd.
-  metadata:   base64 encoded issuance metadata
-  issuance:   YAML encode issuance list. Note: numerical values must be written in strings.
-              eg: '[{recipient: "xrn:sdgkjhs2345u79ghisodg", tradable_amount: "10", retired_amount: "2", retirement_location: "YY-ZZ 12345"}]'
-              Note: "tradable_amount" and "retired_amount" default to 0.
-              Note: "retirement_location" is only required when "retired_amount" is positive.`,
-		Args: cobra.ExactArgs(6),
+  issuer:           issuer address
+  class_id:         credit class
+  start_date:       The beginning of the period during which this credit batch
+                    was quantified and verified. Format: yyyy-mm-dd.
+  end_date:         The end of the period during which this credit batch was
+                    quantified and verified. Format: yyyy-mm-dd.
+  project_location: The location of the project that is backing the credits in
+                    this batch
+  metadata:         base64 encoded issuance metadata
+  issuance:         YAML encode issuance list.
+                    eg: '[{recipient: "xrn:sdgkjhs2345u79ghisodg", tradable_amount: "10", retired_amount: "2", retirement_location: "YY-ZZ 12345"}]'
+                    Note: numerical values must be written in strings.
+                    Note: "tradable_amount" and "retired_amount" default to 0.
+                    Note: "retirement_location" is only required when
+		          "retired_amount" is positive.`,
+		Args: cobra.ExactArgs(0),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			startDate, err := parseDate("start_date", args[2])
+			issuer, err := cmd.Flags().GetString(FlagIssuer)
```

Comment:
I think issuer can just reuse the --from flag now...

Reply:
Is it possible that issuer is not the msg signer?

Reply:
No